src/shared-api-objects/user-has-access-react.component.tsx

```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import { getCurrentUser, userHasAccess } from "./current-user";
+
+export default function UserHasAccessReact(props) {
+  const [user, setUser] = React.useState(null);
+
+  React.useEffect(() => {
+    const subscription = getCurrentUser({ includeAuthStatus: false }).subscribe(
+      x => {
+        setUser(x);
+        console.log(x);
+      }
+    );
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, []);
+
+  if (user) {
+    return userHasAccess(props.privilege, user) !== undefined && props.children;
+  } else {
+    return null;
```

Comment:
no change needed, but here's some fun stuff to learn about: "React Suspense"

It allows you to "suspend" rendering of a component until a promise resolves with some data. It's essentially just another pattern for doing what you've done here with the if/else. You can read more about it here: https://medium.com/@baphemot/understanding-react-suspense-1c73b4b0b1e6